core/src/main/java/hudson/model/DependencyGraph.java

```diff
@@ -213,13 +212,13 @@ private DependencyGraph(boolean dummy) {
     private List<Dependency> get(Map<AbstractProject, List<DependencyGroup>> map, AbstractProject src) {
         List<DependencyGroup> v = map.get(src);
         if(v==null) {
-            return ImmutableList.of();
+            return Collections.emptyList();
         } else {
-            ImmutableList.Builder<Dependency> builder = ImmutableList.builder();
+            List<Dependency> builder = new ArrayList<>();
             for (DependencyGroup dependencyGroup : v) {
                 builder.addAll(dependencyGroup.getGroup());
             }
-            return builder.build();
+            return Collections.unmodifiableList(builder);
```

Comment:
`builder` was declared within this method and goes out of scope at the end of the method, so nothing else can modify it. We do not need to copy it.